django/models/README.md

```diff
@@ -0,0 +1,243 @@
+# Model
+
+Best practices for creating model class.
+
+## Table of Contents
+
+- [Common](#common)
+- [Fields](#fields)
+- [Model Controller](#model-controller)
+- [Managers](#managers)
+
+## Common
+
+These are some of the best practices that you should add to your model class.
+
+### The String Representation Function
+
+By default django will use id when we try to print the object.
+
+```python
+
+obj = Model.objects.first()
+print(obj)
+>>> <Model: (1)>
+```
+
+It's easier to debug when you get more information regarding the object.
+
+```python
+class Person(models.Model):
+    first_name = models.CharField(max_length=50)
+    last_name = models.CharField(max_length=50)
+    position = models.ForeignKey(
+        to='people.Position',
+        on_delete=models.CASCADE,
+        related_name='people',
+    )
+
+    # String function that you should add
+    def __str__(self):
+        return f'{self.first_name} {self.last_name}'
+```
+
+When trying to debug:
+
+```python
+person = Person.objects.first()
+print(person)
+>>> <Person: Khemanorak Khath>
+```
+
+**CAUCTION**: You should not use foreign key field in your string function,
+because this could lead to performance problem.
+
+```python
+# No
+def __str__(self):
+    return f'{self.first_name} {self.last_name} {self.position.name}'
+
+# If your query is not select_related position, this will hit database
+# with the amount of people.
+```
+
+## Fields
+
+Best practices when defining field in model.
+
+### Naming
+
+- Field name should be **noun** and in **singular** form, except
+for many-to-many field should be **plural** form.
+- Boolean field should be **verb** with past tense.
+- Do not repeat model name in field name.
+- Related name parameter value should be **plural** form.
+
+```python
+class Person(models.Model):
+    # Yes: Signular noun
+    first_name = models.CharField(max_length=100)
+    # No : Repeat model name person
+    person_first_name = models.CharField(max_length=100)
+    
+    # Yes: Many to Many in plural form
+    companies = model.ManyToMany(to='company.Company', blank=True, related_name='people')
+
+    # Yes
+    activated = models.BooleanField()
+    # No
+    is_active = models.BooleanFiled()
+```
+
+### Relationship
+
+- When defining foreign key, one to one or many to many in a model it should be
+lazy reference instead of importing the model class directly, preventing
+recursive import exception.
+[Read more](https://docs.djangoproject.com/en/3.1/ref/models/fields/#foreignkey)
+- Always add related name parameter to foreign key or many to many field.
```

Comment:
I don't think we always need to add a related name na krub. Sometimes the model that we referenced won't gonna reversed getting this model. And I also think the performance might be better (even if a little bit 555). Maybe we can suggest others to put `related_name='+'` if they do not want to let Django create a backward relation krub. What do you think krub?

Reply:
I do think we should na, if fk added I think I will be use, and `_set` look ugly.

Reply:
But is it a bad idea to put `related_name='+'` krub? Sometimes I don't want a backward relation so I put `+` to let Django ignore a backward relation krubb.

https://docs.djangoproject.com/en/3.1/ref/models/fields/#django.db.models.ForeignKey.related_name

Reply:
Both of you are correct krub. We should always define the related name but it can be '+' or something.

Reply:
Okay, will add krub. @earthpyy Could you help explain in which case you want to ignore backward relation na krub?